lib/em/protocols/linetext2.rb

```diff
@@ -57,6 +57,7 @@ def receive_data data
         while remaining_data.length > 0
           if @lt2_mode == :lines
             if ix = remaining_data.index( @lt2_delimiter )
+              delimiter_length = remaining_data.slice(@lt2_delimiter).length
```

Comment:
I agree that it is inefficient, but I am not sure how to do this with only one run of the regex.  The code I replaced also had a separate index call and length call.